### PR TITLE
[norelease] create CI workflows

### PR DESCRIPTION
**Pull request by brettinternet**
## Change Type

* [ ] Feature
* [x] Chore
* [ ] Bug Fix

## Change Level

* [ ] major
* [ ] minor
* [ ] patch

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+jobs:
+  version:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.BOT_PAT }} # Hack: A PAT from an admin account to override our branch protections
```
Comment (brettinternet) [inline .github/workflows/publish.yml:15]: Before we merge, we need to add this secret from the PDQJenkins Github account

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup Node
+      uses: actions/setup-node@v1
+      with:
+        node-version: '10.x'
+
+    - name: Publish
+      run: npm publish
```
Comment (brettinternet) [inline .github/workflows/publish.yml:48]: Is there more to this command?

Is there anything else that needs to occur here? I'm guessing the obvious is we're going to need to give access to our NPM publishing.

Comment (TomPridham) [reply .github/workflows/publish.yml:48]: i think we will at least need to do `npm login` prior to that. i can't think of any other steps besides that tho

Comment (brettinternet) [reply .github/workflows/publish.yml]: We can do it with tokens instead

**Comment by brettinternet** [comment]
This should be ready now

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - uses: actions/checkout@v2
+
+    - name: Setup Node
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+        registry-url: 'https://registry.npmjs.org'
+
+    - name: Publish (NPM)
+      run: npm run publish
```
Comment (TomPridham) [inline .github/workflows/publish.yml:46]:
```suggestion
      run: npm publish
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  verify:
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, 'skipci')"
```
Comment (TomPridham) [inline .github/workflows/ci.yml:11]:
```suggestion
    if: "!contains(github.event.head_commit.message, 'norelease')"
```
i think, to match what's in the version script

Comment (brettinternet) [reply .github/workflows/ci.yml:11]: `npm-verison.sh` already checks for this. I was thinking we could have it still run with `norelease` to verify the build, but perhaps we don't need this with the CI running on each commit to a PR

Comment (brettinternet) [reply .github/workflows/ci.yml:11]: Oops, actually because this is `ci.yml`, this won't apply. The CI should always run unless `skipci` is passed.

**Pull request merged**